src/core.jl

```diff
@@ -7,7 +7,13 @@
 # using Sugar
 using LinearAlgebra
 using Statistics
-import Base: CodeInfo, Slot
+import Base: CodeInfo
+
+if VERSION <= v"1.10"
+    import Base: Slot
+else
+    import Core.Compiler.SlotNumber as Slot
+end
```

Comment:
This does not work on old Julia versions where `as` was not defined. For instance in v1.3:

```julia
ERROR: LoadError: LoadError: syntax: extra token "as" after end of expression
```

A possible solution is to raise the requirements for the Julia version, but now that this change was released, old versions would still install the current release and hence be broken. I think the better fix is to use backward-compatible syntax:

```julia
if VERSION <= v"1.10"
    import Base: Slot
else
    import Core.Compiler.SlotNumber
    const Slot = SlotNumber
end
```

Reply:
@schillic Indeed! Fixed it in #56. Thanks for noticing!